pygmt/tests/test_solar.py

```diff
@@ -0,0 +1,101 @@
+"""
+Tests for solar.
+"""
+import datetime
+
+import pytest
+from pygmt import Figure
+from pygmt.exceptions import GMTInvalidInput
+
+
+@pytest.mark.mpl_image_compare
+def test_solar_set_terminator_datetime_string():
+    """
+    Test passing the solar argument with the civil terminator and a datetime
+    string.
+    """
+    fig = Figure()
+    fig.solar(
+        region="d",
+        projection="W0/15c",
+        frame="a",
+        terminator="civil",
+        terminator_datetime="1990-02-17 04:25:00",
+    )
+    return fig
+
+
+@pytest.mark.mpl_image_compare
+def test_solar_default_terminator():
+    """
+    Test passing the solar argument with a time string and no terminator type
+    to confirm the default terminator type.
+    """
+    fig = Figure()
+    fig.solar(
+        region="d",
+        projection="W0/15c",
+        frame="a",
+        terminator_datetime="1990-02-17 04:25:00",
+    )
+    return fig
+
+
+@pytest.mark.mpl_image_compare
+def test_solar_set_terminator_datetime_object():
+    """
+    Test passing the solar argument with the day_night terminator and a
+    datetime string.
+    """
+    fig = Figure()
+    terminator_datetime = datetime.datetime(
+        year=1990, month=2, day=17, hour=4, minute=25, second=0
+    )
+    fig.solar(
+        R="d",
+        J="W0/15c",
+        B="a",
+        termiantor="day_night",
```

Comment:
Typo. Surprised that the tests pass actually, It might be that the wrong image is produced so please double check that you didn't push an incorrect image up onto DAGsHub.
```suggestion
        terminator="day_night",
```

Reply:
Nice catch! `terminator`'s default value is `d`, so even we have a typo here, the baseline image is still correct. 

The typo is directly passed to GMT as a invalid arguments `-termiantorday_night`, and GMT actually gives a warning:
```
solar [WARNING]: Transparency is expected in percentage.  Did you mean 0?
```

It actually shows the big limitation of how PyGMT handles arguments (as in #256).